From 07f179eb96fa51ce9e764293eb1373291f4c3f3d Mon Sep 17 00:00:00 2001
From: John Zhuge <john.zhuge@gmail.com>
Date: Mon, 15 Aug 2016 17:23:19 -0700
Subject: [PATCH 1754/2848] HDFS-8682. Should not remove decommissioned
 node,while calculating the number of live/dead
 decommissioned node. (Contributed by J. Andreina)

(cherry picked from commit fdb56f74f38cabb0f94e0781fcedb1594904c026)
(cherry picked from commit 2371ed0551ec71e5b7a9de39645861b9db6f6903)

Change-Id: Ia8617b91dc106943fc871e9275547e89a76ca11e
---
 .../hadoop/hdfs/server/namenode/FSNamesystem.java  |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java
index 08b168f..504dea9 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java
@@ -6798,7 +6798,7 @@ public int getNumDeadDataNodes() {
   @Override // FSNamesystemMBean
   public int getNumDecomLiveDataNodes() {
     final List<DatanodeDescriptor> live = new ArrayList<DatanodeDescriptor>();
-    getBlockManager().getDatanodeManager().fetchDatanodes(live, null, true);
+    getBlockManager().getDatanodeManager().fetchDatanodes(live, null, false);
     int liveDecommissioned = 0;
     for (DatanodeDescriptor node : live) {
       liveDecommissioned += node.isDecommissioned() ? 1 : 0;
@@ -6809,7 +6809,7 @@ public int getNumDecomLiveDataNodes() {
   @Override // FSNamesystemMBean
   public int getNumDecomDeadDataNodes() {
     final List<DatanodeDescriptor> dead = new ArrayList<DatanodeDescriptor>();
-    getBlockManager().getDatanodeManager().fetchDatanodes(null, dead, true);
+    getBlockManager().getDatanodeManager().fetchDatanodes(null, dead, false);
     int deadDecommissioned = 0;
     for (DatanodeDescriptor node : dead) {
       deadDecommissioned += node.isDecommissioned() ? 1 : 0;
-- 
1.7.9.5

